src/it/java/org/eclipse/microprofile/starter/TestMatrixTest.java

```diff
@@ -0,0 +1,448 @@
+                testWeb(urlBase + MPSpec.JWT_AUTH.urlContent[0][0], 5, MPSpec.JWT_AUTH.urlContent[0][1]);
+                testWeb(urlBase + c + MPSpec.OPEN_API.urlContent[0][0], 5, MPSpec.OPEN_API.urlContent[0][1]);
+                testWeb(urlBase + MPSpec.REST_CLIENT.urlContent[0][0], 5, MPSpec.REST_CLIENT.urlContent[0][1]);
+            } else if (specSelection == SpecSelection.ALL_BUT_JWT_REST) {
+                // Verify content
+                testWeb(urlBase + MPSpec.DEFAULT.urlContent[0][0], 5, MPSpec.DEFAULT.urlContent[0][1]);
+                testWeb(urlBase + MPSpec.CONFIG.urlContent[0][0], 5, MPSpec.CONFIG.urlContent[0][1]);
+                testWeb(urlBase + MPSpec.CONFIG.urlContent[1][0], 5, MPSpec.CONFIG.urlContent[1][1]);
+                testWeb(urlBase + MPSpec.FAULT_TOLERANCE.urlContent[0][0], 5, MPSpec.FAULT_TOLERANCE.urlContent[0][1]);
+                testWeb(urlBase + c + MPSpec.HEALTH_CHECKS.urlContent[0][0], 5, MPSpec.HEALTH_CHECKS.urlContent[0][1]);
+                testWeb(urlBase + MPSpec.METRICS.urlContent[0][0], 5, MPSpec.METRICS.urlContent[0][1]);
+                testWeb(urlBase + c + MPSpec.METRICS.urlContent[1][0], 5, MPSpec.METRICS.urlContent[1][1]);
+                testWeb(urlBase + c + MPSpec.OPEN_API.urlContent[0][0], 5, MPSpec.OPEN_API.urlContent[0][1]);
+            } else if (specSelection == SpecSelection.JWT_REST) {
+                // Verify content
+                testWeb(urlBase + MPSpec.DEFAULT.urlContent[0][0], 5, MPSpec.DEFAULT.urlContent[0][1]);
+                testWeb(urlBase + MPSpec.JWT_AUTH.urlContent[0][0], 5, MPSpec.JWT_AUTH.urlContent[0][1]);
+                testWeb(urlBase + MPSpec.REST_CLIENT.urlContent[0][0], 5, MPSpec.REST_CLIENT.urlContent[0][1]);
+            } else {
+                throw new IllegalArgumentException(
+                        "Unexpected SpecSelection enum value. Have you updated SpecSelection?");
+            }
+
+            LOGGER.info("Terminate and scan logs...");
+            // This is a move that makes it flush on my Linux x86_64 OpenJDK J9/HotSpot 11. Does no harm on Win.
+            int bytes = pA.getInputStream().available();
+            if (specSelection.hasServiceB) {
+                bytes = pB.getInputStream().available();
+            }
+
+            processStopper(pA, artifactId);
+            if (specSelection.hasServiceB) {
+                processStopper(pB, artifactId);
+            }
+
+            checkLog(this.getClass().getCanonicalName(), testName.getMethodName(), "Runtime log", runLogA);
+            if (specSelection.hasServiceB) {
+                checkLog(this.getClass().getCanonicalName(), testName.getMethodName(), "Runtime log", runLogB);
+            }
+            LOGGER.info("Gonna wait for ports closed...");
+            // Release ports
+            Commands.waitForTcpClosed("localhost", Commands.parsePort(urlBase), 60);
```

Comment:
No verification if socket is really closed (return true) or timed out (return false)